calc/l3_route_resolver.go

```diff
@@ -783,7 +870,17 @@ func (r RouteInfo) IsValidRoute() bool {
 		r.Block.NodeName != "" ||
 		len(r.Host.NodeNames) > 0 ||
 		r.Pool.NATOutgoing ||
-		r.LocalWEP.RefCount > 0
+		len(r.WEPs) > 0
+}
+
+// Copy returns a copy of the RouteInfo. Since some fields are pointers, we need to
+// explicitly copy them so that they are not shared between the copies.
+func (r RouteInfo) Copy() RouteInfo {
+	cp := r
+	if len(r.WEPs) != 0 {
+		cp.WEPs = append(cp.WEPs, r.WEPs...)
```

Comment:
`cp` will have WEPs initialised to the same slice so this will result in `cp.WEPs` being a doubled copy of `r.WEPs` (and it may share backing storage debending on the capacity of `r.WEPs`.)  I think you want:
```
cp.WEPs = make([]WEP, len(r.WEPs))
copy(cp.WEPs, r.WEPs)
```